test/com/esotericsoftware/kryo/serializers/SerializerErrorBehaviorTest.java

```diff
@@ -0,0 +1,49 @@
+package com.esotericsoftware.kryo.serializers;
+
+import com.esotericsoftware.kryo.KryoException;
+import com.esotericsoftware.kryo.KryoTestCase;
+import com.esotericsoftware.kryo.Registration;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayOutputStream;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class SerializerErrorBehaviorTest extends KryoTestCase {
+
+    @Test
+    void testAbstractClassCannotBeSerialized() {
+        kryo.register(AbstractClass.class);
+        kryo.register(ConcreteClass.class);
+        assertThrows(KryoException.class, () -> simpleRoundtrip(AbstractClass.class, new ConcreteClass()));
+    }
+
+    @Test
+    void testCustomObjectInsantiatorForAbstractClass() {
+        Registration registration = kryo.register(AbstractClass.class);
+        registration.setInstantiator(ConcreteClass::new);
+    }
+
+    @Test
+    void testNonMemberAbstractClassCannotBeSerialized() {
+        kryo.register(TestAbstractClass.class);
+        kryo.register(TestConcreteClass.class);
+        assertThrows(KryoException.class, () -> simpleRoundtrip(TestAbstractClass.class, new TestConcreteClass()));
```

Comment:
Please assert on the error message as well, so we known that your changes actually work. Something like:

```java
KryoException thrown = assertThrows(KryoException.class, () -> simpleRoundtrip(TestAbstractClass.class, new TestConcreteClass()));
assertTrue(thrown.getMessage().contains("The type you are trying to serialize into is abstract");
```